subsys/logging/log_core.c

```diff
@@ -125,7 +129,10 @@ static u32_t count_s(const char *str, u32_t nargs)
  */
 static bool is_rodata(const void *addr)
 {
-#if defined(CONFIG_ARM) || defined(CONFIG_ARC) || defined(CONIFG_RISCV32) || \
+#if defined(CONFIG_ARM)
+	#define RO_START _CODE_BASE_ADDR
```

Comment:
This changes the logic a bit: Here you are using code region to detect rodata, while other architectures relies on the firmware image boundaries. So behaviour of the function will change between architectures. For example pointer to data stored in flash will be "rodata" on the ARM but not on other platforms. Are we OK with that? If so, I would suggest adding a comment here.

Reply:
i can add a comment. As library example shown relying on sections is not 100% sure. Fixed memory map is better, however i don't know if other architectures has fixed addresses like ARM. However, once #16681 gets in, then it will better to use log_strdup whenever there is even slight possibility as it won't be duplicated unless ro data.